### PR TITLE
encrypt GH auth token in database

### PR DESCRIPTION
**Pull request by mwrock**
Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
related to #3378

File: components/builder-jobsrv/src/server/worker_manager.rs [hunk 1]
```diff
@@ -212,6 +212,20 @@ impl WorkerMgr {
...
+                match bldr_core::integrations::decrypt(
+                    &self.key_dir,
+                    job.get_project().get_vcs_auth_token(),
+                ) {
+                    Ok(secret) => job.mut_project().set_vcs_auth_token(secret),
+                    Err(e) => {
+                        debug!("failed to decrypt vcs auth token, err={:?}", e);
+                        job.set_state(jobsrv::JobState::Pending);
+                        self.datastore.update_job(&job)?;
+                        continue;
```
Comment (christophermaier) [inline components/builder-jobsrv/src/server/worker_manager.rs]: How will this scenario eventually be handled? It seems like we'll just be perpetually trying to decrypt something we can't decrypt; won't that hold up the entire group? What actions can be taken to remediate this?

**Comment by mwrock** [comment]
Good calls @christophermaier . I have addressed both points.

**Comment by mwrock** [comment]
closing since we moved awauy from personal access tokens

**Pull request closed without merging**